### PR TITLE
removed CC054 support and replaced it with support for CC014

### PR DESCRIPTION
**Pull request by RobertLaverick**

### CODE REVIEW

File: app/models/DepartureStatus.scala [hunk 1]
```diff
@@ -50,11 +50,12 @@ object DepartureStatus extends Enumerable.Implicits with MongoDateTimeFormats {
...
-      case MessageReceivedEvent.ControlDecisionNotification => ControlDecisionNotification
-      case MessageReceivedEvent.NoReleaseForTransit         => NoReleaseForTransit
-      case MessageReceivedEvent.ReleaseForTransit           => ReleaseForTransit
-      case _                                                => throw new Exception(s"Tried to transition from MrnAllocated to $messageReceived.")
+      case MessageReceivedEvent.MrnAllocated                   => MrnAllocated
+      case MessageReceivedEvent.ControlDecisionNotification    => ControlDecisionNotification
+      case MessageReceivedEvent.NoReleaseForTransit            => NoReleaseForTransit
+      case MessageReceivedEvent.ReleaseForTransit              => ReleaseForTransit
+      case MessageReceivedEvent.DeclarationCancellationRequest => DeclarationCancellationRequest
+      case _                                                   => throw new Exception(s"Tried to transition from MrnAllocated to $messageReceived.")
```
Comment (crystalmark) [inline app/models/DepartureStatus.scala:58]: Will this message be returned to the 3rd party app?  If so, it could do with some Matthew loving.

**Review by crystalmark — COMMENTED** [review]
Could the conf/xsd/CC054A.xsd file be removed now?

**Comment by jhkldev** [comment]
> Could the conf/xsd/CC054A.xsd file be removed now?

👍

File: app/models/DepartureStatus.scala (hunk 1 above)
Comment (RobertLaverick) [reply app/models/DepartureStatus.scala:58]: In this case no, as I've just realised we're not using the transition on the public side. Do we want to ensure we are?

Comment (RobertLaverick) [reply app/models/DepartureStatus.scala:58]: I'm updating the code to at least force it to use the transition here, that way we at least get the exception. If we complete the ticket John put together for using an Either, we will see the message, at the moment we don't.

**Review by crystalmark — APPROVED** [review]

**Review by jhkldev — APPROVED** [review]

**Pull request merged**